backend/pkg/api/connect/service/kafkaconnect/defaulter.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package kafkaconnect
+
+import v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+
+// Defaulter updates a given kafkaconnect request with defaults.
+type defaulter struct{}
+
+func (*defaulter) applyListConnectorsRequest(req *v1alpha1.ListConnectorsRequest) {
+	if req.GetPageSize() == 0 {
+		req.PageSize = 100
+	}
+}
```

Comment:
I do wonder if we should attempt at a generic solution?

Quickly trying, something like this seems to work:

```go

type defaulterTest struct{}

type PageSizer interface {
	GetPageSize() int32
}

func (*defaulterTest) applyDefaultPageSizeToRequest(m proto.Message) {
	if psMsg, ok := m.(PageSizer); ok {
		if psMsg.GetPageSize() == 0 {
			messageDesc := m.ProtoReflect().Descriptor()
			pageSizeField := messageDesc.Fields().ByName("page_size")
			if pageSizeField != nil {
				m.ProtoReflect().Set(pageSizeField, protoreflect.ValueOfInt32(100))
			}
		}
	}
}

func TestPageSizeDefaulter(t *testing.T) {
	pzd := defaulterTest{}
	msg0 := &v1alpha1.ListTopicsRequest{}
	msg1 := &v1alpha1.ListTopicsRequest{PageSize: 10}
	pzd.applyDefaultPageSizeToRequest(msg0)
	pzd.applyDefaultPageSizeToRequest(msg1)

	assert.Equal(t, int32(100), msg0.GetPageSize())
	assert.Equal(t, int32(10), msg1.GetPageSize())
}
```

Reply:
Generic version:

```go
func TestPageSizeDefaulter(t *testing.T) {
	msg0 := &v1alpha1.ListTopicsRequest{}
	msg1 := &v1alpha1.ListTopicsRequest{PageSize: 10}
	applyDefaultPageSizeToRequest(msg0)
	applyDefaultPageSizeToRequest(msg1)

	assert.Equal(t, int32(100), msg0.GetPageSize())
	assert.Equal(t, int32(10), msg1.GetPageSize())
}

type PageSizerT interface {
	proto.Message

	GetPageSize() int32
}

func applyDefaultPageSizeToRequest[T PageSizerT](m T) {
	if m.GetPageSize() == 0 {
		messageDesc := m.ProtoReflect().Descriptor()
		pageSizeField := messageDesc.Fields().ByName("page_size")
		if pageSizeField != nil {
			m.ProtoReflect().Set(pageSizeField, protoreflect.ValueOfInt32(100))
		}
	}
}
```

Reply:
I think it was mostly by accident that all default page sizes are 100, we will likely have different defaults and page sizes for each endpoint in the future. Personally I currently favour this approach, so that you can see all the applied defaults for a request at the some place, but I'll keep this in mind.